### DISCUSSION TITLE
How to deploy tailwindcss with html on github (using npm)?

### DISCUSSION THREAD

**Discussion by SheikhElMoctarG** [tailwindlabs/tailwindcss#12024] [Help] [answered]
Hi,

I have a project tailwind html, but i used npm to install tailwindcss, but when I deployed it on github pages, tailwindcss not working
[demo](https://sheikhelmoctarg.github.io/bookmark/)

can you help me ?

**Comment by iazaran** [comment, marked as answer]
Please check these 5 steps in your project: https://tailwindcss.com/docs/installation

**Reply by wongjn** [reply]
I do not believe this is relevant to their problem since it seems like they have installed Tailwind correctly for use locally. Looking at the [GitHub Actions run](https://github.com/SheikhElMoctarG/bookmark/actions/runs/6212376922/job/16862443558), I believe the problem is that the GitHub Pages deployment steps have not been set up to run a Tailwind build.

**Reply by iazaran** [reply]
If we used `npm run build` in GitHub Pages configuration, then we need to check those steps, specially `<link href="/dist/output.css" rel="stylesheet">` in index.html.
But yes, as you said, otherwise, it is related to the GitHub Pages configuration.

**Reply by wongjn** [reply]
`/dist/output.css` would not be the correct path, since this would resolve to `https://sheikhelmoctarg.github.io/dist/output.css`, whereas they are hosting on a subdirectory and thus the generated CSS would be reachable at `https://sheikhelmoctarg.github.io/bookmark/dist/output.css`. I believe they do not need to modify the `<link>` if they generate the CSS into `css/output.css` like they do in their `run:tail` `npm` script.

**Reply by iazaran** [reply]
Exactly, they are using a path to CSS, now it is `./css/output.css`.
But if they are using `npm run build` in GitHub Pages configuration, then they will see something like this in the `<head>`: `<link rel="stylesheet" href="/bookmark/assets/index-ff56465r.css">`.
So I think the problem is in GitHub Pages configuration, as you said.
My suggestion is running `npm run build` locally, checking the built files, and then config the GitHub Pages configuration, based on observations.

**Reply by wongjn** [reply]
`npm run build` would not work in GitHub pages since they have no `build` script defined in their `package.json`. I believe they do not need to modify the `<link>` if they generate the CSS into `css/output.css` like they do in their `run:tail` `npm` script, perhaps with a `build` script like:

```
npx tailwindcss -i css/input.css -o css/output.css
```

**Reply by iazaran** [reply]
Yes :+1: 
Step 4 in https://tailwindcss.com/docs/installation

**Reply by wongjn** [reply]
Not quite, step 4 has:
```
npx tailwindcss -i ./src/input.css -o ./dist/output.css --watch
```
This would output to `dist/output.css`, not `css/output.css` (although they could change the `<link>` tag `href` to `./dist/output.css`, though they would presumably need to make a similar change in their existing `run:tail` script). Also `src/input.css` does not exist in their repository so I'd presume they would need to move their `css/input.css` to `src/input.css` (presumably, they'd need to make a similar change in their existing `run:tail` script again). Furthermore, the `--watch` flag would mean the process does not exit, meaning the GitHub workflow would not complete and thus the deployment would not take place. Perhaps consider a `build` script like:

```
npx tailwindcss -i css/input.css -o css/output.css
```

So that they would not need to make any further modifications.

**Reply by iazaran** [reply]
You are right :+1: 
I mentioned the step in there, not the paths, also --watch isn't needed.

**Reply by SheikhElMoctarG** [reply]
That's working now ✅
my problem was I used locally method for intalling tailwindcss, but when I change it to cdn, everything works fine.
thank you for your help.

**Reply by wongjn** [reply]
CDN is not recommended for production. You may wish to consider perusing [the documentation](https://docs.github.com/en/pages) to set up GitHub pages properly for your project. If you need more assistance with this, consider seeking assistance in an area more related to GitHub pages.

**Reply by SheikhElMoctarG** [reply]
do you have a tutorial? can you give me some links explain it ?

**Reply by iazaran** [reply]
Github pages: https://docs.github.com/en/pages

Tailwind: https://tailwindcss.com/docs/installation

You can find tutorials on the web, too, but the main documentation is always the best.

**Comment by wongjn** [comment]
Looking at the [GitHub Actions run](https://github.com/SheikhElMoctarG/bookmark/actions/runs/6212376922/job/16862443558), it seems like you have not set up GitHub Pages deployment to run a Tailwind build. Consider perusing [the documentation](https://docs.github.com/en/pages) to set up GitHub pages properly for your project. If you need more assistance with this, consider seeking assistance in an area more related to GitHub pages.

**Comment by erenworld** [comment]
Hi everyone!
My Tailwind CSS isn’t applying for github pages with the new version. Any ideas?
Repo link : https://github.com/erenworld/personal-site

**Reply by zenzen26** [reply]
Hi have you found the solutions? My website looks perfect find on local but the css file is always 404 in gh pages. Since there is no config files in v4. How do you do it?

I set up like 
```
npx create-next-app@latest reponame --typescript --eslint (... yes for src tailwind postcss etc)
cd reponame
npm install --save-dev gh-pages

```

next.config.ts
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  output: 'export',
  distDir: 'out',
  basePath: isProd ? 'reponame' : '',
  assetPrefix: isProd ? 'reponame' : '',
  images: {
    unoptimized: true,
  },
};

module.exports = nextConfig;
```

package.json
```
"scripts": {
  "dev": "next dev",
  "build": "next build",
  "start": "next start",
  "export": "next export",
  "deploy": "next build && next export && gh-pages -d out -b gh-pages"
}

```
